cpu/efm32/periph/timer.c

```diff
@@ -52,18 +65,48 @@ static inline bool _is_wtimer(timer_t dev)
 #endif
 }
 
-int timer_init(tim_t dev, unsigned long freq, timer_cb_t callback, void *arg)
+/**
+ * @brief   Check whether dev is using a LETIMER device
+ */
+static inline bool _is_letimer(timer_t dev)
 {
-    TIMER_TypeDef *pre, *tim;
+#if defined(LETIMER_COUNT) && LETIMER_COUNT > 0
+    return ((uint32_t) timer_config[dev].timer.dev) == LETIMER0_BASE;
+#else
+    (void) dev;
+    return false;
+#endif
+}
 
-    /* test if given timer device is valid */
-    if (dev >= TIMER_NUMOF) {
-        return -1;
+static void _letimer_init(tim_t dev, unsigned long freq)
+{
+    if (freq != 32768) {
+        LOG_ERROR("letimer only supports 32768 Hz\n");
+        assert(0);
     }
 
-    /* save callback */
-    isr_ctx[dev].cb = callback;
-    isr_ctx[dev].arg = arg;
+    LETIMER_TypeDef *tim = timer_config[dev].timer.dev;
+
+    /* use external LF crystal */
+    CMU_ClockSelectSet(cmuClock_LFA, cmuSelect_LFXO);
```

Comment:
This would be awful if true. I'm looking at my one disassembled tradfri device and indeed I don't suppose I see an LF crystal. I don't know who deletes a part critical for low-power functionality just to save $0.10. This is disappointing.

Reply:
Is it that crucial? Using the HFRCO would certainly result in a worse clock in terms of accuracy, but it's not like such clocks wouldn't be expected in an IoT setting, and radio-timing specifics are managed by whichever oscillator is used for radio conditioning anyway.

Reply:
It's crucial for sleeping for long periods of time (minutes or more) while still being able to wake with ~0.5 ms accuracy so as not to waste energy powering the radio for several ms to check for a beacon packet. This is why virtually all IoT radio modules include both the HF crystal (for the RF) and the LF crystal (for sleep timing).

But if that's what they gave us to work with then that's what we'll make work.

Reply:
I just realized the clock source is already set in the board configurations so I'll just drop this line anyway.